LongestPole/Rectangles.lean

```diff
@@ -0,0 +1,83 @@
+/-
+Copyright (c) 2024 Lean FRO. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kim Morrison
+-/
+
+/-!
+# Finding the maximal rectangles in a list of points.
+-/
+
+structure Rectangle where
```

Comment:
```suggestion
/--
A rectangle `r` is a product of the left-closed, right-open intervals `[r.left, r.right) × [r.bottom, r.top)`,
whose corners are indexed by natural numbers.
-/
structure Rectangle where
```

Reply:
I could alternatively use a product of `Ico`s, but I presume you considered this already.

Reply:
Another option: use the integers for indexing; this would simplify some checks here (but presumably at the expense of other checks in `Unused`.

Reply:
> /--
> A rectangle `r` is a product of the left-closed, right-open intervals `[r.left, r.right) × [r.bottom, r.top)`,
> whose corners are indexed by natural numbers.
> -/

Didn't want any Mathlib imports here.